# Cannoli Changelog #

### Version 0.3.0 ###

**Closed issues:**

 - Add implicit methods that attach to source RDD [\#131](https://github.com/bigdatagenomics/cannoli/issues/131)
 - Flip function and command line class names around [\#130](https://github.com/bigdatagenomics/cannoli/issues/130)
 - Add API documentation link and badge [\#128](https://github.com/bigdatagenomics/cannoli/issues/128)
 - Add homebrew formula at brewsci/homebrew-bio [\#124](https://github.com/bigdatagenomics/cannoli/issues/124)
 - Add bioconda recipe [\#123](https://github.com/bigdatagenomics/cannoli/issues/123)
 - Support validation stringency in out formatters [\#122](https://github.com/bigdatagenomics/cannoli/issues/122)
 - Add Ensembl Variant Effect Predictor (VEP) for variant annotation [\#112](https://github.com/bigdatagenomics/cannoli/issues/112)
 - Add Minimap2 for alignment [\#111](https://github.com/bigdatagenomics/cannoli/issues/111)

**Merged and closed pull requests:**

 - Update release script for changelog. [\#143](https://github.com/bigdatagenomics/cannoli/pull/143) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-141] Update ADAM dependency to 0.25.0. [\#142](https://github.com/bigdatagenomics/cannoli/pull/142) ([heuermh](https://github.com/heuermh))
 - Update default docker image for bowtie2. [\#140](https://github.com/bigdatagenomics/cannoli/pull/140) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-138] Update Cannoli per latest ADAM snapshot changes. [\#139](https://github.com/bigdatagenomics/cannoli/pull/139) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-131] Add implicits on Cannoli function source data sets. [\#133](https://github.com/bigdatagenomics/cannoli/pull/133) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-130] Extract function classes to core package. [\#132](https://github.com/bigdatagenomics/cannoli/pull/132) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-128] Adding API documentation link and badge. [\#129](https://github.com/bigdatagenomics/cannoli/pull/129) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-112]  Adding Ensembl Variant Effect Predictor (VEP) for variant annotation [\#127](https://github.com/bigdatagenomics/cannoli/pull/127) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-122] Support validation stringency in out formatters. [\#126](https://github.com/bigdatagenomics/cannoli/pull/126) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-111] Adding Minimap2 for alignment. [\#119](https://github.com/bigdatagenomics/cannoli/pull/119) ([heuermh](https://github.com/heuermh))



### Version 0.2.0 ###

**Closed issues:**

 - Update ADAM dependency version to 0.24.0. [\#118](https://github.com/bigdatagenomics/cannoli/issues/118)
 - Javadoc error and warnings [\#115](https://github.com/bigdatagenomics/cannoli/issues/115)
 - Update pipe method calls due to latest ADAM 0.24.0 snapshot [\#114](https://github.com/bigdatagenomics/cannoli/issues/114)
 - Split commands with subcommands into separate Cannoli CLI classes [\#110](https://github.com/bigdatagenomics/cannoli/issues/110)
 - Jenkins build failing due to upstream changes. [\#108](https://github.com/bigdatagenomics/cannoli/issues/108)
 - Provide functions for use in cannoli-shell or notebooks. [\#104](https://github.com/bigdatagenomics/cannoli/issues/104)
 - Error running BWA with Docker [\#103](https://github.com/bigdatagenomics/cannoli/issues/103)
 - Allow use of Singularity instead of Docker [\#98](https://github.com/bigdatagenomics/cannoli/issues/98)
 - Bump ADAM dependency version to 0.24.0-SNAPSHOT. [\#95](https://github.com/bigdatagenomics/cannoli/issues/95)
 - Drop support for Scala 2.10 and Spark 1.x. [\#94](https://github.com/bigdatagenomics/cannoli/issues/94)
 - Tidy up FreeBayes [\#67](https://github.com/bigdatagenomics/cannoli/issues/67)
 - Support loading reference files from HDFS/other file system [\#50](https://github.com/bigdatagenomics/cannoli/issues/50)
 - Attributes from freebayes header missing from variants and genotypes [\#43](https://github.com/bigdatagenomics/cannoli/issues/43)
 - Factor out docker/mapping code [\#34](https://github.com/bigdatagenomics/cannoli/issues/34)
 - Add wrappers for GMAP and GSNAP aligners [\#29](https://github.com/bigdatagenomics/cannoli/issues/29)
 - Jenkins failures due to missing publish_scaladoc.sh [\#21](https://github.com/bigdatagenomics/cannoli/issues/21)

**Merged and closed pull requests:**

 - [CANNOLI-118] Update ADAM dependency version to 0.24.0. [\#121](https://github.com/bigdatagenomics/cannoli/pull/121) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-110] Split commands with subcommands into separate Cannoli CLI classes. [\#117](https://github.com/bigdatagenomics/cannoli/pull/117) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-115] Fix javadoc error and warnings. [\#116](https://github.com/bigdatagenomics/cannoli/pull/116) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-108] Command argument to pipe is now Seq[String]. [\#109](https://github.com/bigdatagenomics/cannoli/pull/109) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-98] Adding container builder. [\#107](https://github.com/bigdatagenomics/cannoli/pull/107) ([heuermh](https://github.com/heuermh))
 - Allow Singularity to run containers [\#106](https://github.com/bigdatagenomics/cannoli/pull/106) ([jpdna](https://github.com/jpdna))
 - [CANNOLI-95] Bump ADAM dependency version to 0.24.0-SNAPSHOT [\#102](https://github.com/bigdatagenomics/cannoli/pull/102) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-94] Dropping support for Scala 2.10 and Spark 1.x. [\#101](https://github.com/bigdatagenomics/cannoli/pull/101) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-94][CANNOLI-95] Drop support for Scala 2.10 and Spark 1.x. [\#100](https://github.com/bigdatagenomics/cannoli/pull/100) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-43] Use accumulator for VCF header lines. [\#72](https://github.com/bigdatagenomics/cannoli/pull/72) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-104] Provide functions for use in cannoli-shell or notebooks. [\#69](https://github.com/bigdatagenomics/cannoli/pull/69) ([heuermh](https://github.com/heuermh))
 - Add CannoliCommand and CannoliAlignerCommand. [\#54](https://github.com/bigdatagenomics/cannoli/pull/54) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-29] Add minimal GMAP and GSNAP wrappers. [\#32](https://github.com/bigdatagenomics/cannoli/pull/32) ([heuermh](https://github.com/heuermh))


### Version 0.1.0 ###

**Closed issues:**

 - Default Docker images to those on BioContainers registry [\#92](https://github.com/bigdatagenomics/cannoli/issues/92)
 - Bump to ADAM 0.23.0 [\#89](https://github.com/bigdatagenomics/cannoli/issues/89)
 - Jenkins compile error due to upstream 0.23.0-SNAPSHOT changes [\#87](https://github.com/bigdatagenomics/cannoli/issues/87)
 - Error with snpEff [\#86](https://github.com/bigdatagenomics/cannoli/issues/86)
 - Add wrapper for RealTimeGenomics/rtg-tools vcfeval [\#84](https://github.com/bigdatagenomics/cannoli/issues/84)
 - Incorrect case for snpEff when using docker [\#83](https://github.com/bigdatagenomics/cannoli/issues/83)
 - Cannoli-BWA Command Failure // Re Open [\#78](https://github.com/bigdatagenomics/cannoli/issues/78)
 - Cannoli-BWA Command Failure [\#77](https://github.com/bigdatagenomics/cannoli/issues/77)
 - All commands invoked via docker should use --rm=true [\#76](https://github.com/bigdatagenomics/cannoli/issues/76)
 - FreeBayes Issue [\#75](https://github.com/bigdatagenomics/cannoli/issues/75)
 - Read in Fragment not found [\#74](https://github.com/bigdatagenomics/cannoli/issues/74)
 - Cannoli Command [\#73](https://github.com/bigdatagenomics/cannoli/issues/73)
 - FileNotFoundException while loading file from HDFS [\#66](https://github.com/bigdatagenomics/cannoli/issues/66)
 - Create script for releasing to Sonatype OSS repository [\#63](https://github.com/bigdatagenomics/cannoli/issues/63)
 - Compile errors with latest ADAM 0.23.0-SNAPSHOT version [\#55](https://github.com/bigdatagenomics/cannoli/issues/55)
 - Rename FastqInterleaver --> InterleaveFastq [\#51](https://github.com/bigdatagenomics/cannoli/issues/51)
 - Remove SequenceDictionaryReader per ADAMContext.loadSequenceDictionary [\#48](https://github.com/bigdatagenomics/cannoli/issues/48)
 - Move adam-cli to provided scope [\#46](https://github.com/bigdatagenomics/cannoli/issues/46)
 - NoSuchMethodError when running cannoli locally [\#45](https://github.com/bigdatagenomics/cannoli/issues/45)
 -  BCF not yet supported  [\#44](https://github.com/bigdatagenomics/cannoli/issues/44)
 - Interleaver should support uBAM [\#39](https://github.com/bigdatagenomics/cannoli/issues/39)
 - Use --interleaved for bowtie2 [\#36](https://github.com/bigdatagenomics/cannoli/issues/36)
 - Use ADAM tab5 formatter for bowtie [\#33](https://github.com/bigdatagenomics/cannoli/issues/33)
 - Add ability to specify docker command [\#28](https://github.com/bigdatagenomics/cannoli/issues/28)
 - Bump BWA docker version argument [\#27](https://github.com/bigdatagenomics/cannoli/issues/27)
 - BWA-Mount index files into docker container [\#25](https://github.com/bigdatagenomics/cannoli/issues/25)
 - Disable incremental mode in the Scala build [\#23](https://github.com/bigdatagenomics/cannoli/issues/23)
 - Update plugin dependency versions to match bigdatagenomics/adam [\#20](https://github.com/bigdatagenomics/cannoli/issues/20)
 - Bowtie error: reads file does not look like a FASTQ file [\#18](https://github.com/bigdatagenomics/cannoli/issues/18)
 - Update ADAM dependency version to 0.22.0 [\#16](https://github.com/bigdatagenomics/cannoli/issues/16)
 - Add ability to force load format for BWA [\#14](https://github.com/bigdatagenomics/cannoli/issues/14)
 - Add a FASTQ interleaver [\#12](https://github.com/bigdatagenomics/cannoli/issues/12)
 - Properly handle sequence and read group metadata from BWA [\#10](https://github.com/bigdatagenomics/cannoli/issues/10)
 - Set up CI [\#9](https://github.com/bigdatagenomics/cannoli/issues/9)
 - Need to double escape tabs in BWA native [\#7](https://github.com/bigdatagenomics/cannoli/issues/7)
 - Migrate repository to bigdatagenomics organization [\#6](https://github.com/bigdatagenomics/cannoli/issues/6)
 - Allow BWA to save Fragments instead of AlignmentRecords [\#4](https://github.com/bigdatagenomics/cannoli/issues/4)

**Merged and closed pull requests:**

 - [CANNOLI-92] Use BioContainers Docker images with version tags by default. [\#96](https://github.com/bigdatagenomics/cannoli/pull/96) ([heuermh](https://github.com/heuermh))
 - Add hadoopConfiguration to VCFOutFormatter ctr. [\#93](https://github.com/bigdatagenomics/cannoli/pull/93) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-89] Bump ADAM dependency version to 0.23.0. [\#90](https://github.com/bigdatagenomics/cannoli/pull/90) ([heuermh](https://github.com/heuermh))
 - Fix compile error due to upstream 0.23.0-SNAPSHOT changes. [\#88](https://github.com/bigdatagenomics/cannoli/pull/88) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-83] Fix case for snpEff when using docker. [\#85](https://github.com/bigdatagenomics/cannoli/pull/85) ([heuermh](https://github.com/heuermh))
 - Replace cannoli ascii logo. [\#82](https://github.com/bigdatagenomics/cannoli/pull/82) ([heuermh](https://github.com/heuermh))
 - Add docker support to bowtie and bowtie2. [\#81](https://github.com/bigdatagenomics/cannoli/pull/81) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-76] Add --rm flag to docker run. [\#80](https://github.com/bigdatagenomics/cannoli/pull/80) ([heuermh](https://github.com/heuermh))
 - Adding code of contact adapted from the Contributor Convenant, version 1.4. [\#79](https://github.com/bigdatagenomics/cannoli/pull/79) ([heuermh](https://github.com/heuermh))
 - Adding wrapper for BCFtools norm. [\#71](https://github.com/bigdatagenomics/cannoli/pull/71) ([heuermh](https://github.com/heuermh))
 - Adding wrapper for samtools mpileup. [\#70](https://github.com/bigdatagenomics/cannoli/pull/70) ([heuermh](https://github.com/heuermh))
 - Adding wrapper for vt normalize. [\#68](https://github.com/bigdatagenomics/cannoli/pull/68) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-45] Move to multimodule Maven build, add cannoli-submit script [\#62](https://github.com/bigdatagenomics/cannoli/pull/62) ([heuermh](https://github.com/heuermh))
 - Update move_to scripts for Spark version 2.1.0. [\#61](https://github.com/bigdatagenomics/cannoli/pull/61) ([heuermh](https://github.com/heuermh))
 - Fix compile error in FragmentRDD.apply [\#60](https://github.com/bigdatagenomics/cannoli/pull/60) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-33] Use ADAM tab5 formatter for bowtie. [\#59](https://github.com/bigdatagenomics/cannoli/pull/59) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-48] Remove SequenceDictionaryReader. [\#58](https://github.com/bigdatagenomics/cannoli/pull/58) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-55] Fix compile errors due to upstream commit 0afbda96. [\#57](https://github.com/bigdatagenomics/cannoli/pull/57) ([heuermh](https://github.com/heuermh))
 - Fix compile issues with most recent ADAM version 0.23.0-SNAPSHOT. [\#56](https://github.com/bigdatagenomics/cannoli/pull/56) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-51] Rename FastqInterleaver --> InterleaveFastq [\#53](https://github.com/bigdatagenomics/cannoli/pull/53) ([heuermh](https://github.com/heuermh))
 - Code style fixes. [\#52](https://github.com/bigdatagenomics/cannoli/pull/52) ([heuermh](https://github.com/heuermh))
 - Update adam dependency version to 0.23.0-SNAPSHOT. [\#49](https://github.com/bigdatagenomics/cannoli/pull/49) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-46] Move adam-cli to provided scope. [\#47](https://github.com/bigdatagenomics/cannoli/pull/47) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-33] Use ADAM tab5 formatter for bowtie [\#42](https://github.com/bigdatagenomics/cannoli/pull/42) ([heuermh](https://github.com/heuermh))
 - Adding sample reads tool. [\#41](https://github.com/bigdatagenomics/cannoli/pull/41) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-39] Add support for exporting interleaved FASTQ as unaligned BAM. [\#40](https://github.com/bigdatagenomics/cannoli/pull/40) ([fnothaft](https://github.com/fnothaft))
 - Use --interleaved for bowtie2 [\#37](https://github.com/bigdatagenomics/cannoli/pull/37) ([heuermh](https://github.com/heuermh))
 - Add placeholder publish_scaladoc.sh script [\#31](https://github.com/bigdatagenomics/cannoli/pull/31) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-28] Allow docker command to be specified. [\#30](https://github.com/bigdatagenomics/cannoli/pull/30) ([fnothaft](https://github.com/fnothaft))
 - [CANNOLI-25] Properly support loading BWA indices via SparkFiles API. [\#26](https://github.com/bigdatagenomics/cannoli/pull/26) ([fnothaft](https://github.com/fnothaft))
 - Disable incremental mode in the Scala build. [\#24](https://github.com/bigdatagenomics/cannoli/pull/24) ([heuermh](https://github.com/heuermh))
 - Update examples in readme [\#22](https://github.com/bigdatagenomics/cannoli/pull/22) ([heuermh](https://github.com/heuermh))
 - Prep for move to bigdatagenomics org [\#19](https://github.com/bigdatagenomics/cannoli/pull/19) ([fnothaft](https://github.com/fnothaft))
 - [CANNOLI-16] Update ADAM dependency version to 0.22.0 [\#17](https://github.com/bigdatagenomics/cannoli/pull/17) ([heuermh](https://github.com/heuermh))
 - [CANNOLI-14] Add switches to force loading from Interleaved FASTQ/Parquet. [\#15](https://github.com/bigdatagenomics/cannoli/pull/15) ([fnothaft](https://github.com/fnothaft))
 - [CANNOLI-12] Add a FASTQ interleaver. [\#13](https://github.com/bigdatagenomics/cannoli/pull/13) ([fnothaft](https://github.com/fnothaft))
 - [CANNOLI-10] Attach record groups and sequence dictionary to BWA output. [\#11](https://github.com/bigdatagenomics/cannoli/pull/11) ([fnothaft](https://github.com/fnothaft))
 - [CANNOLI-7] Escape tabs in @RG line when running BWA without docker. [\#8](https://github.com/bigdatagenomics/cannoli/pull/8) ([fnothaft](https://github.com/fnothaft))
 - [CANNOLI-4] Add ability to save Fragments from BWA. [\#5](https://github.com/bigdatagenomics/cannoli/pull/5) ([fnothaft](https://github.com/fnothaft))
 - Add Docker arguments to Freebayes and SnpEff. [\#3](https://github.com/bigdatagenomics/cannoli/pull/3) ([heuermh](https://github.com/heuermh))
 - Add BWA runner. [\#2](https://github.com/bigdatagenomics/cannoli/pull/2) ([fnothaft](https://github.com/fnothaft))
 - SupportedHeaderLines renamed to DefaultHeaderLines [\#1](https://github.com/bigdatagenomics/cannoli/pull/1) ([waltermblair](https://github.com/waltermblair))
